### PR TITLE
P/stieg/osx support

### PR DESCRIPTION
**Pull request by stieg**
Adds the ability for us to compile the unit test from OSX.  Achieves this by moving our unit tests to a 64bit platform and by adding support to work with the clang compiler on OSX.

Next I want to move our unit tests over to clang and begin to use their static analyzer on our code so that we are able to find bugs in the firmware.  Plus moving to clang will make it easier not to break the mac environment.

### CODE REVIEW

File: src/logger/loggerApi.c [hunk 1]
```diff
@@ -643,7 +643,7 @@ int api_getAnalogConfig(Serial *serial, const jsmntok_t * json){
 			startIndex = endIndex = modp_atoi(json->data);
 		}
 	}
-	if (startIndex >= 0 && startIndex <= CONFIG_ADC_CHANNELS){
+	if (startIndex <= CONFIG_ADC_CHANNELS){
 		sendAnalogConfig(serial, startIndex, endIndex);
```
Comment (brentpicasso) [inline src/logger/loggerApi.c]: This is unrelated optimization to the OSX thing, right?

**Comment by brentpicasso** [comment]
some of this looks like general clean up and optimizations, which is good by itself. how much of it was to get OSX support working?

File: src/logger/loggerApi.c (hunk 1 above)
Comment (stieg) [inline src/logger/loggerApi.c]: Its an unsigned variable where we are checking if its value is less than 0.  So more of cleanup.  CLANG was complaining about it.  Its not needed.

**Comment by stieg** [comment]
All of it was needed to get OSX working with our unit tests.

File: src/logger/loggerApi.c (hunk 1 above)
Comment (brentpicasso) [inline src/logger/loggerApi.c]: I saw that, looks like a sane change :+1:
